railties/lib/rails/test_unit/runner.rb

```diff
@@ -58,19 +58,19 @@ def compose_filter(runnable, filter)
 
         private
           def extract_filters(argv)
-            argv.select { |arg| arg =~ /^\w+\// }.map do |path|
+            argv.select { |arg| arg =~ %r{^/?\w+/} }.map do |path|
               case
               when path =~ /(:\d+)+$/
                 file, *lines = path.split(":")
                 filters << [ file, lines ]
                 file
               when Dir.exist?(path)
                 "#{path}/**/*_test.rb"
-              else
+              when path !~ /\/$/
                 filters << [ path, [] ]
                 path
               end
-            end
+            end.compact
```

Comment:
So matching for a leading slash also matches against regex matchers passed to the `-n` option since regex is passed in `/like this/`. I explicitly filter them out in the final `when` by not matching a trailing slash. This works because the directory matching happens before this so no directories are passed to the final condition.

These  regexes don't match anything and are mapped as `nil`. So we need to discard them.